polygon/heimdall/event_record.go

```diff
@@ -28,9 +28,60 @@ type EventRecordWithTime struct {
 	Time time.Time `json:"record_time" yaml:"record_time"`
 }
 
+func (e *EventRecordWithTime) toIntTime() eventRecordWithIntTime {
+	return eventRecordWithIntTime{
+		EventRecord: EventRecord{
+			ID:       e.ID,
+			Contract: e.Contract,
+			Data:     e.Data,
+			TxHash:   e.TxHash,
+			LogIndex: e.LogIndex,
+			ChainID:  e.ChainID,
+		},
+		Time: big.NewInt(e.Time.Unix()),
+	}
+}
+
+func (e *EventRecordWithTime) EncodeRLP() (rlp.RawValue, error) {
+	r := e.toIntTime()
+	return rlp.EncodeToBytes(&r)
+}
+
+// DecodeEventRecord RLP decodes the given bytes to EventRecordWithTime
+func DecodeEventRecord(v rlp.RawValue) (*EventRecordWithTime, error) {
+	var event eventRecordWithIntTime
+	err := rlp.DecodeBytes(v, &event)
+	if err != nil {
+		return nil, err
+	}
+
+	e := event.toTime()
+
+	return &e, nil
+}
+
+type eventRecordWithIntTime struct {
```

Comment:
can you please help me understand why we need all of this and we can't use the already existing approach of persisting and reading state sync events which is:
- storing the packed data from calling `stateReceiverABI.Pack("commitState", big.NewInt(eventRecord.Time.Unix()), recordBytes)` as in https://github.com/ledgerwatch/erigon/blob/main/eth/stagedsync/bor_heimdall_shared.go#L481-L488
- simply reading that as bytes like here https://github.com/ledgerwatch/erigon/blob/main/turbo/snapshotsync/freezeblocks/block_reader.go#L1303

Reply:
Simplified RLP encoding to use `stateReceiverABI.Pack`